src/core/meshcop/timestamp.cpp

```diff
@@ -35,54 +35,78 @@
 
 #include "common/code_utils.hpp"
 #include "common/num_utils.hpp"
+#include "common/numeric_limits.hpp"
 
 namespace ot {
 namespace MeshCoP {
 
-void Timestamp::ConvertTo(otTimestamp &aTimestamp) const
+void Timestamp::ConvertTo(Info &aInfo) const
 {
-    aTimestamp.mSeconds       = GetSeconds();
-    aTimestamp.mTicks         = GetTicks();
-    aTimestamp.mAuthoritative = GetAuthoritative();
+    aInfo.mSeconds       = GetSeconds();
+    aInfo.mTicks         = GetTicks();
+    aInfo.mAuthoritative = GetAuthoritative();
 }
 
-void Timestamp::SetFromTimestamp(const otTimestamp &aTimestamp)
+void Timestamp::SetFrom(const Info &aInfo)
 {
-    SetSeconds(aTimestamp.mSeconds);
-    SetTicks(aTimestamp.mTicks);
-    SetAuthoritative(aTimestamp.mAuthoritative);
+    SetSeconds(aInfo.mSeconds);
+    SetTicks(aInfo.mTicks);
+    SetAuthoritative(aInfo.mAuthoritative);
 }
 
-int Timestamp::Compare(const Timestamp *aFirst, const Timestamp *aSecond)
+void Timestamp::SetToInvalid(void)
 {
-    int rval;
+    mSeconds16       = NumericLimits<uint16_t>::kMax;
+    mSeconds32       = NumericLimits<uint32_t>::kMax;
+    mTicksAndAutFlag = NumericLimits<uint16_t>::kMax;
```

Comment:
nit: easier to read "Authoritative"?
```suggestion
    mTicksAndAuthFlag = NumericLimits<uint16_t>::kMax;
```

Reply:
Good idea. Thanks. Changed in new push.